frontend/src/stores/useTemplateStore.ts

```diff
@@ -0,0 +1,27 @@
+import { create, SetState } from 'zustand'
+
+import {
+  Template
+} from '../types/ProjectTypes'
+
+interface TemplateStore {
+  template: Template,
+  setTemplate: (template: Template) => void
+  // update: (template: Template) => void,
+  // setName: (newName: string) => void
```

Comment:
```suggestion
  // TODO: Support updating/renaming
```
Just so we can find what needs to be done by searching for `// TODO`